noir-projects/noir-contracts/contracts/nft_contract/src/main.nr

```diff
@@ -134,21 +135,29 @@ contract NFT {
         public_owners_storage.write(to);
     }
 
+    // Transfers token with `token_id` from public balance of message sender to a private balance of `to`.
+    #[private]
+    fn transfer_to_private(to: AztecAddress, token_id: Field) {
+        let from = context.msg_sender();
+
+        let nft = NFT::at(context.this_address());
+
+        // We prepare the transfer.
+        let hiding_point_slot = nft.prepare_transfer_to_private(from, to).call(&mut context);
```

Comment:
Great idea. We should investigate whether it would make sense to support calling all the functions as internal because I can imagine that this pattern will be very common. Do you agree with me creating an issue for macros and linking it in the code?

Reply:
Addressed in https://github.com/AztecProtocol/aztec-packages/pull/9150/commits/920aa4b96433d4c334c17711b7aa762b94b0d26b